### PR TITLE
fix: make the library compatible with AoT and universal, dont bundle stylesheets

### LINKED ISSUE #550: AOT builds are failiing because of the ngx-chips dependency... I believe the culprit is the version of angular-cli you are using.
PLEASE MAKE SURE THAT:
- you searched similar issues online (9/10 issues in this repo are solved by googling, so please...do it)
- you provide an online demo I can see without having to replicate your environment
- you help me by debugging your issue, and if you can't, do go on filling out this form

**I'm submitting a ...**  (check one with "x")
```
[ x] bug report => search github for a similar issue or PR before submitting
[ ] support request/question

Notice: feature requests will be ignored, submit a PR if you'd like
```

**Current behavior**

In my case the bug is manifesting because Angular AOT is run via ngtools/webpack which is the sister of angular-cli which you are using.  So this is what is happening... Because this dependency 
is trying to run its build and AOT upon installing the package i.e. into the dist folder <<< it is running against my webpack config's ngtoosl/webpack installation housing its AOT parameters.  thus causing the AOT build to fail

Below I state recreation steps but the issue is definatelty stemming from how you're building the package upon installation which is using AOT parameters from anguar cli.  

My first thought is... why place this step on the user installing a package... I get if someone wanted to customize the css or files but having this step run alongside of another person's AOT installation and params should / could be thought out better so that this won't occur.  

I stated in another thread here for this package is that AOT ngtools / angular-cli was having issues with one of their dependencies... it was auto incremented through ^ and broke a lot of peoples AOT builds. something with enhanced-resolver...  What's insane is that they did a fixed version and have since then reverted back to allowing ^ increments for minor updates.

I would like to work through this with you as my build depends on it.  

**Expected behavior**

**Minimal reproduction of the problem with instructions (if applicable)**

Download an Angular quick repo and use ngtools/webpack inside of it... then install your package and you will see the error.  

**What do you use to build your app?. Please specify the version**

**Angular version:** 

**ngx-chips version:**

**Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]

### PR DESCRIPTION
**Pull request by mattlewis92**
Same changes as in https://github.com/Gbuomprisco/ng2-material-dropdown/pull/37, with `ng2-material-dropdown` pointed to a temporary fork until that PR can be reviewed and merged.

BREAKING CHANGE:
The libraries stylesheet must now be included somewhere in your app. For CLI users this is a case of adding this line to `src/styles.css`:
```
@import '~ngx-chips/dist/ngx-chips.css';
```

Fixes #550, #537, #497, #492, #451

### CODE REVIEW

**Comment by xtianus79** [comment]
Ahhh you good sir!!!! I might be loving you today

**Comment by mattlewis92** [comment]
Happy to help 😄 If you want to use this now you can `npm i @mattlewis92/ngx-chips`

**Comment by xtianus79** [comment]
@mattlewis92 the only thing I question is the taking out the inlining of the stylesheets.  but before i make judegment i have a question.  

I used the scss mixin's to create a custom template.  lol what is going to happen to that now?

**Comment by mattlewis92** [comment]
My reasoning behind that was to make it easier to override the CSS - in my app I have to use `!important` on every css rule I was to overwrite. The scss is [still there](https://unpkg.com/@mattlewis92/ngx-chips@2.0.1/dist/modules/tag-input.module.scss) and you can import it into your app to create your own theme. In fact that might actually decrease your bundle size a bit as then you only import what you need.

**Comment by xtianus79** [comment]
cool let me give it a spin... yes that is why i used the mixins an deep as to not have to override.  if you nest your css to a more specific order than the style you usually won't have to do !important i.e.

     this > dom > tree > element > style class {
        background: blue;
    } 

will beat 

    style class {
      ...

**Comment by mattlewis92** [comment]
That's true, but that feels even hackier + more verbose than using `!important`. Also by inlining the stylesheets into the components you wouldn't be able to use sass and hence its mixins. This is the only option I can see that will work 😞

**Comment by xtianus79** [comment]
i am getting this error with your repo 

Could not resolve ngx-chips relative to /Users/xxxx/xxx/xxxxx/frontend/src/app/app.module.ts.

**Comment by Gbuomprisco** [comment]
Hi @mattlewis92, thanks for taking the time to work on this. The objections I have are related to the amount of changes needed for this, and practices I don't agree with like using any, or inlining everything in components. I am quite sure we can find a way to fix the issue without these changes. As for  the breaking change, that would require a version bump too.

TL;DR: I am not going to merge this for now. I'll have a look this weekend to see if we can fix this in another way. But thanks!

**Comment by mattlewis92** [comment]
I absolutely agree with you that I wish there was a better way to do this. If you wish to keep the template + style sheet urls (and having the lib easily consumable and just work without any additional config by the end user) you will have to go down the route of post processing your template and metadata files to inline the sources. This is what the material team do in order for it to work for them:
https://github.com/angular/material2/blob/master/tools/package-tools/inline-resources.ts
https://github.com/angular/material2/blob/master/tools/package-tools/metadata-inlining.ts

In my opinion, that adds even more unnecessary complexity that what I've proposed in this PR, for the sake of just inlining templates and having the user include a stylesheet. 

If you want to discuss more let me know, I'd love to work together with you on this to find an elegant solution 😄

**Comment by xtianus79** [comment]
@mattlewis92 I have used your repo and got my build working AOT with minimal effort (some css went a little wonky).  Nevertheless, I definatly appreciate your hard work.  I will say this at least puts everyone in the right direction.  

Thanks @Gbuomprisco for looking into this and hopefully something more concrete comes along this weekend for you.

**Comment by xtianus79** [comment]
@mattlewis92 hey buddy.  I have your fork installed and was wondering why the close icon had dissapeared?  Is that something I need to pull in i.e. like the fonts?
